azerothcore/azerothcore-wotlk
# feat(Core/Formations): implemented a new formation groupAI flag "GROUP_AI_FLAG_EVADE_TOGETHER"

This PR implements a new groupAI flag which will allow formation member which is evading to trigger evade for other members too

## Changes Proposed:
-  Implements a new creature_formation groupAI flag - "GROUP_AI_FLAG_EVADE_TOGETHER"

## Issues Addressed:

- Closes 

## SOURCE:


## Tests Performed:

- Tested with Golemagg encounter (Molten Core)
- 


## How to Test the Changes:


For tests you gonna need least 2x players. Apply GROUP_AI_FLAG_EVADE_TOGETHER (4) on attackable formation group (preferring for all attackable members). Engage members that formation members (would be best if you move them a little to sides to see actual effect a bit better) and at one of them attempt to trigger evade (would be easier if single formation member wouldn't have threat on both characters). Thats it ;)

## Known Issues and TODO List:


None so far...


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
